buildbot/configure.py

```diff
@@ -103,10 +94,28 @@ def do_configure(args):
         llvm_build_shared_libs = 'ON'
 
     if args.use_lld:
-      llvm_enable_lld = 'ON'
+        llvm_enable_lld = 'ON'
 
-    if build_libclc:
-        llvm_enable_projects += ';libclc'
+    # CI Default conditionally appends to options, keep it at the bottom of
+    # args handling
+    if args.ci_defaults:
+        print("#############################################")
+        print("# Default CI configuration will be applied. #")
+        print("#############################################")
+
+        # For clang-format and clang-tidy
+        llvm_enable_projects += ";clang-tools-extra"
+        # libclc is required for CI validation
+        if 'libclc' not in llvm_enable_projects:
+            llvm_enable_projects += ';libclc'
+        # libclc passes `--nvvm-reflect-enable=false`, build NVPTX to enable it
+        if 'NVPTX' not in llvm_targets_to_build:
+            llvm_targets_to_build += ';NVPTX'
+        # Add both NVIDIA and AMD libclc targets
+        if libclc_amd_target_names not in libclc_targets_to_build:
+            libclc_targets_to_build += libclc_amd_target_names
```

Comment:
I think to build libclc for AMD target, LLVM's AMDGPU target back-end must be built. Am I right?

Reply:
That was my guess as well, but I've just tried and `LIBCLC_TARGETS_TO_BUILD:STRING=;amdgcn--;amdgcn--amdhsa` builds fine with only `LLVM_TARGETS_TO_BUILD:STRING=X86`. Saying that I'm worried that if in future libclc uses more AMD specific code (for example builtins) it will not be possible. Enforcing AMDGPU backend seems like a wise thing to do moving forward.